src/db/DbThreadStore.ts

```diff
@@ -0,0 +1,121 @@
+/*
+Copyright 2019 mx-puppet-bridge
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+
+import {IDatabaseConnector} from "mx-puppet-bridge/lib/src/db/connector";
+import {Log} from "mx-puppet-bridge/lib/src";
+
+const log = new Log("DbThreadStore");
+
+export class DbThreadStore {
+    constructor(
+        private db: IDatabaseConnector,
+    ) {
+    }
+
+    public async setFirstThreadEvent(eventId: string, firstThreadEvent: string) {
+        let query: string;
+        if (this.db.type == "postgres") {
+            query = `INSERT INTO thread_store 
+                                (event_id, thread_first_event_id) 
+                            VALUES (
+                                    $eventId, 
+                                    $first
+                                )   
+                            ON CONFLICT (event_id) DO UPDATE
+                                SET thread_first_event_id = $first`;
+        } else {
+            query = `REPLACE INTO thread_store 
+                                (event_id, thread_first_event_id, thread_last_event_id) 
+                                VALUES (
+                                    $eventId, 
+                                    $first, 
+                                    SELECT thread_last_event_id FROM thread_store WHERE event_id = $eventId
+                                )`;
+        }
+
+        await this.db.Run(query, {
+            eventId: eventId,
+            first: firstThreadEvent
+        });
+    }
+
+    public async setLastThreadEvent(eventId: string, lastThreadEvent: string) {
+        let query: string;
+        if (this.db.type == "postgres") {
```

Comment:
Maybe instead of checking for type you could check for existance and delete first, that way we could add other SQLs down the road more easily

Reply:
I could, but this is kinda throwing away capabilities of the database system we if we have to do it manually.